tests/iperf-vsock/build_iperf_vsock.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# This file is part of Canonical's TDX repository which includes tools
+# to setup and configure a confidential computing environment
+# based on Intel TDX technology.
+# See the LICENSE file in the repository for the license text.
+
+# Copyright 2024 Canonical Ltd.
+# SPDX-License-Identifier: GPL-3.0-only
+
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranties
+# of MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+
+if [ ! -f "${SCRIPT_DIR}/../iperf/bin/iperf3" ]; then
+    echo "Cloning and building iperf-vsock"
+    rm -rf ${SCRIPT_DIR}/../iperf
+    TMP_DIR=$(mktemp -d)
+    pushd ${TMP_DIR}
+    git clone https://github.com/stefano-garzarella/iperf-vsock.git --branch iperf-vsock-3.9
```

Comment:
I'm not sure what the alternative is here? Also all the tox tests are using python packages from PyPI which are installed over the network.

Reply:
Yea, its either keep the binary executable prebuilt or do this.  I think this is the right approach.  It will only build once.  Much of what we do requires network access (apt update, any apt install...)

Reply:
You could also put it in a PPA or snap it. On our squad, we use [a PPA](https://launchpad.net/~canonical-nvidia/+archive/ubuntu/perftest+cuda) to hold custom binary builds consumed by our tests.

Reply:
I moved this to a PPA as it seems like a really good solution